### PR TITLE
samples: restore cpp_synchronization test

### PR DESCRIPTION
**Pull request by tyrantlizard77**
Issue: ZEP-2172
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by lpereira — CHANGES_REQUESTED** [review]
LGTM, but please s/fiber|task/thread/

**Comment by andrewboie** [comment]
should be fixed now

File: samples/cpp_synchronization/README.txt [hunk 1]
```diff
@@ -0,0 +1,62 @@
+Title: C++ Synchronization
+
+Description:
+The sample project illustrates usage of pure virtual class, member
+functions with different types of arguments, global objects constructor
+invocation.
+
+A simple application demonstrates basic sanity of the kernel.
+The background task and a fiber take turns printing a greeting message to the
```
Comment (nashif) [inline samples/cpp_synchronization/README.txt]: task and fiber?

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by andrewboie** [comment]
README updated

**Review by nashif — APPROVED** [review]

**Pull request merged**